gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/processor/SynchronizationService.java

```diff
@@ -48,6 +50,23 @@ public <T> boolean checkSynchronization(final Class<T> entityClass, final T depl
                 boolean previousAccessibleState = f.isAccessible();
                 f.setAccessible(true);
                 try {
+                    // FIXME dirty hack to ignore null pointer and empty object comparisions on Properties & Services
```

Comment:
just a thought. How about ensuring properties default to an empty list if null ?

I think that would keep this part of the code cleaner

e.g.

```java
public class Properties implements Serializable {

    @JsonProperty("properties")
    private List<Property> properties = List.of();

    @JsonIgnore
    private Map<String, String> entries = Map.of();

    public void setProperties(List<Property> properties) {
        this.properties = properties == null ? List.of() : properties;

        this.entries = this.properties.stream()
                .collect(
                    Collectors.toMap(
                        Property::getKey,
                        Property::getValue,
                        (v1, v2) -> {
                            throw new RuntimeException(String.format("Duplicate key for values %s and %s", v1, v2));
                        },
                        TemplatedValueHashMap::new
                    )
                );
    }

    public List<Property> getProperties() {
        return properties;
    }

    @JsonIgnore
    public Map<String, String> getValues() {
        return this.entries;
    }
}
```

Reply:
Also, I ran a test with empty service and cannot replicate the issue

Reply:
And maybe for this specific situation (null vs empty properties) we could add a unit test ?